manager/allocator/cnmallocator/networkallocator.go

```diff
@@ -404,6 +404,9 @@ func (na *cnmNetworkAllocator) IsServiceAllocated(s *api.Service, flags ...func(
 	vipLoop:
 		for _, vip := range s.Endpoint.VirtualIPs {
 			if na.IsVIPOnIngressNetwork(vip) && networkallocator.IsIngressNetworkNeeded(s) {
+				if _, ok := na.services[s.ID]; !ok {
```

Comment:
Please add the comment here explaining the scenario which is , If the ingress network is required and the allocation is not done for the ingress network.